pkg/issuer/acme/dns/util/wait.go

```diff
@@ -115,15 +118,64 @@ func checkAuthoritativeNss(fqdn, value string, nameservers []string) (bool, erro
 				}
 			}
 		}
-
-		if !found {
+	}
+	if !found {
+		if len(errs) > 0 {
+			return false, fmt.Errorf("%v", errs)
+		} else {
 			return false, nil
 		}
 	}
 
 	return true, nil
 }
 
+//// checkAuthoritativeNss queries each of the given nameservers for the expected TXT record.
+//func checkAuthoritativeNss(fqdn, challenge string, nameservers []string) (bool, error) {
+//	var errs []error
+//	for _, ns := range nameservers {
+//		var nsErrs []error
+//		r, err := dnsQuery(fqdn, dns.TypeTXT, []string{net.JoinHostPort(ns, "53")}, false)
+//		if err != nil {
+//			nsErrs = append(nsErrs, err)
+//			continue
+//		}
+//
+//		// NXDomain response is not really an error, just waiting for propagation to happen
+//		if !(r.Rcode == dns.RcodeSuccess || r.Rcode == dns.RcodeNameError) {
+//			 nsErrs = append(nsErrs, fmt.Errorf("NS %s returned %s for %s", ns, dns.RcodeToString[r.Rcode], fqdn))
+//			 continue
+//		}
+//
+//		glog.V(6).Infof("Querying TXT records for %q from server %q", fqdn, ns)
+//		var found bool
+//		for _, rr := range r.Answer {
+//			if txt, ok := rr.(*dns.TXT); ok {
+//				if strings.Join(txt.Txt, "") == challenge {
+//					glog.V(6).Infof("TXT record(s) found [%s] from server %q", challenge, ns)
+//					found = true
+//					break
+//				}
+//			}
+//		}
+//
+//		if !found {
+//			if (len(nsErrs) > 0) {
+//				errorf := fmt.Errorf("%v", nsErrs)
+//				glog.V(6).Infof("Errors encountered searching TXT record %q from server %q: %v", fqdn, ns, errorf)
+//				errs = append(errs, nsErrs...)
+//			} else {
+//				glog.V(6).Infof("No TXT record found for %q from server %q", fqdn, ns)
+//			}
+//			continue
+//		} else {
+//			break
+//		}
+//	}
+//
+//	return true, nil
+//}
```

Comment:
Don't think this commented function was meant to be included?

Reply:
Yes, I was leaving it in for review, I thought I made it clearer in the original commit logs that this was a draft commit, apologies!

The code is related to the change in semantics of all servers / one server. It makes more sense now and I'll clean it up with the other changes.